KnowledgeExtractor/WikipediaKnowledgeGraphExtractor.cs

```diff
@@ -96,34 +101,70 @@ public static KnowledgeGraph ExtractKnGraphFromUris(List<UriAndOriginalGraphType
         public static KnowledgeGraph GetKnowledgeGraphFromWikipedia()
         {
             KnowledgeGraph result = new KnowledgeGraph();
-
-            // get the kn graph from the uris
             result = ExtractKnGraphFromUris(WikipediaPagesToParse);
+            AddEdgesBetweenAlgorithmsAndDataStructures(ref result);
+            return result;
+        }
+
+        public static void AddEdgesBetweenAlgorithmsAndDataStructures(ref KnowledgeGraph graph)
+        {
+            List<WordCount> dataStructureWordsInGraph = GetDataStructureWordsInEachWikiPage();
+            Dictionary<string, KnGNode> dataStructureNodes = GetDataStructureNodesWithoutDuplicates(graph);
```

Comment:
I noticed you un-abbreviated some things, which is generally good. Consider also changing KnGNode to KnowledgeGraphNode